internal/controller/suite_test.go

```diff
@@ -67,27 +80,53 @@ var _ = BeforeSuite(func() {
 		BinaryAssetsDirectory: filepath.Join("..", "..", "bin", "k8s",
 			fmt.Sprintf("1.29.0-%s-%s", runtime.GOOS, runtime.GOARCH)),
 	}
-
+	ctx, cancel = context.WithCancel(context.TODO())
 	var err error
 	// cfg is defined in this file globally.
 	cfg, err = testEnv.Start()
 	Expect(err).NotTo(HaveOccurred())
 	Expect(cfg).NotTo(BeNil())
 
+	// Add Policy API scheme
+	err = policyAPI.AddToScheme(scheme.Scheme)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Add Kyverno scheme
+	err = kyvernov1.AddToScheme(scheme.Scheme)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Add Kyverno beta scheme
+	err = kyvernov2beta1.AddToScheme(scheme.Scheme)
+	Expect(err).NotTo(HaveOccurred())
+
 	//+kubebuilder:scaffold:scheme
 
 	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
 	Expect(err).NotTo(HaveOccurred())
 	Expect(k8sClient).NotTo(BeNil())
 
+	k8sManager, err := ctrl.NewManager(cfg, ctrl.Options{
+		Scheme: scheme.Scheme,
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	go func() {
+		defer GinkgoRecover()
+		err = k8sManager.Start(ctx)
+		Expect(err).NotTo(HaveOccurred(), "failed to run manager")
+	}()
+
 })
 
 var _ = AfterSuite(func() {
 	By("tearing down the test environment")
 	if testEnv == nil {
 		return
 	}
-
 	err := testEnv.Stop()
+	if err != nil {
+		time.Sleep(5 * time.Second)
```

Comment:
I don't think this is necessary, at least running locally seems to work without it. You can call `cancel()` above it to stop the manager though.

Reply:
In some cases it gave an error, thinking the stopping of the environment was a bit too slow at times. 
But I'm not able to reproduce that problem, I'll add calling the cancel() function and remove the waiting bit. Thanks